### ISSUE TITLE
Godot Shader error check does not check if inout is const

### ISSUE THREAD

**Issue by danilw** [godotengine/godot#39815]
**Godot version:**
3.2.1 stable

**OS/device including version:**
linux

**Issue description, steps to reproduce:**
Sending constant to inout does not trigger godot shader error check.
Shader code:
```
void func1(inout float val){
	val+=1.;
}

float func2(){
	return 1.;
}
```
and call anywhere `func1(func2());` in shader

**Comment by clayjohn** [comment]
Cc @Chaosus

**Comment by Chaosus** [comment]
Seems like its fixed in the master but not in 3.2 branch

**Comment by akien-mga** [comment]
Fixed by #39820.
